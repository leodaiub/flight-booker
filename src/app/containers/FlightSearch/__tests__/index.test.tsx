import * as React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';

import { configureAppStore } from 'store/configureStore';
import { FlightSearch } from '../';
import { initReactI18next } from 'react-i18next';
import i18next from 'i18next';

const renderComponent = (store: Store) =>
  render(
    <Provider store={store}>
      <HelmetProvider>
        <FlightSearch />
      </HelmetProvider>
    </Provider>,
  );

describe('<FlightSearch />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    i18next.use(initReactI18next);
    store = configureAppStore();
  });
  it('should match the snapshot', () => {
    const component = renderComponent(store);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
