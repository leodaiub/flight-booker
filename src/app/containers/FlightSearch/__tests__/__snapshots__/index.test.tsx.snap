// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightSearch /> should match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
>
  <form>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-12"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        >
          Welcome to your web check-in
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink="false"
            for="filled-basic"
            id="filled-basic-label"
          >
            Flight Number
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              id="filled-basic"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink="false"
            for="filled-basic"
            id="filled-basic-label"
          >
            Full Name
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              id="filled-basic"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Search Flight
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
